libs/wire-subsystems/src/Wire/UserStore/Cassandra.hs

```diff
@@ -11,27 +12,61 @@ import Polysemy.Error
 import Wire.API.User hiding (DeleteUser)
 import Wire.StoredUser
 import Wire.UserStore
+import Wire.UserStore.IndexUser hiding (userId)
 import Wire.UserStore.Unique
 
 interpretUserStoreCassandra :: (Member (Embed IO) r) => ClientState -> InterpreterFor UserStore r
 interpretUserStoreCassandra casClient =
   interpret $
-    runEmbedded (runClient casClient) . \case
+    runEmbedded (runClient casClient) . embed . \case
       GetUser uid -> getUserImpl uid
-      UpdateUser uid update -> embed $ updateUserImpl uid update
-      UpdateUserHandleEither uid update -> embed $ updateUserHandleEitherImpl uid update
-      DeleteUser user -> embed $ deleteUserImpl user
-      LookupHandle hdl -> embed $ lookupHandleImpl LocalQuorum hdl
-      GlimpseHandle hdl -> embed $ lookupHandleImpl One hdl
-      LookupStatus uid -> embed $ lookupStatusImpl uid
-      IsActivated uid -> embed $ isActivatedImpl uid
-      LookupLocale uid -> embed $ lookupLocaleImpl uid
-
-getUserImpl :: (Member (Embed Client) r) => UserId -> Sem r (Maybe StoredUser)
-getUserImpl uid = embed $ do
+      GetIndexUser uid -> getIndexUserImpl uid
+      GetIndexUsersPaginated mPagingState -> getIndexUserPaginatedImpl mPagingState
+      UpdateUser uid update -> updateUserImpl uid update
+      UpdateUserHandleEither uid update -> updateUserHandleEitherImpl uid update
+      DeleteUser user -> deleteUserImpl user
+      LookupHandle hdl -> lookupHandleImpl LocalQuorum hdl
+      GlimpseHandle hdl -> lookupHandleImpl One hdl
+      LookupStatus uid -> lookupStatusImpl uid
+      IsActivated uid -> isActivatedImpl uid
+      LookupLocale uid -> lookupLocaleImpl uid
+
+getUserImpl :: UserId -> Client (Maybe StoredUser)
+getUserImpl uid = do
   mUserTuple <- retry x1 $ query1 selectUser (params LocalQuorum (Identity uid))
   pure $ asRecord <$> mUserTuple
 
+getIndexUserImpl :: UserId -> Client (Maybe IndexUser)
+getIndexUserImpl u = do
+  mIndexUserTuple <- retry x1 $ query1 cql (params LocalQuorum (Identity u))
+  pure $ asRecord <$> mIndexUserTuple
+  where
+    cql :: PrepQuery R (Identity UserId) (TupleType IndexUser)
+    cql = prepared . QueryString $ getIndexUserBaseQuery <> " WHERE id = ?"
+
+getIndexUserPaginatedImpl :: Maybe PagingState -> Client (PageWithState IndexUser)
+getIndexUserPaginatedImpl mPagingState =
+  asRecord <$$> paginateWithState cql (paramsPagingState LocalQuorum () 10000 mPagingState)
```

Comment:
The 10000 might be too much, earlier it was 1000, perhaps allow specifying it using a param.